### PR TITLE
ENG-13560 immediate shutdown of Kafka importer to avoid blocking other processes

### PR DESCRIPTION
**Pull request by zeemanhe**
When cluster state changes, such as @pause, importers will  grab a lock on ImporterManager and will be shutdown. In the meantime, catalog update needs the same lock to proceed. Shutting down importers immediately to avoid blocking catalog update.

### CODE REVIEW

File: src/frontend/org/voltdb/importer/ImportManager.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ private synchronized void restartImporters(Map<String, ImportConfiguration> newP
      * @param catalogContext new catalog context
      * @return new importer configuration
      */
-    private synchronized Map<String, ImportConfiguration> loadNewConfigAndBundles(CatalogContext catalogContext) {
+    private Map<String, ImportConfiguration> loadNewConfigAndBundles(CatalogContext catalogContext) {
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/importer/ImportManager.java]: @zeemanhe I think we need to remove all the synchronized under updateCatalog() method. As the caller already guarantee that there is only one caller.

**Comment by xinjiacs** [comment]
@zeemanhe Does this race deadlock issue start to happen when we upgrade the importer with kafka10?

**Comment by zeemanhe** [comment]
I have not traced down. It is possible since Kafka10 has different shutdown mechanim.

File: src/frontend/org/voltdb/importclient/kafka10/KafkaStreamImporter.java [hunk 2]
```diff
@@ -153,8 +153,9 @@ public void stop() {
 
         if (m_executorService != null) {
             try {
+                //shutdown immediately. Consumers will pick up where they left
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/importclient/kafka10/KafkaStreamImporter.java]: @zeemanhe Can you add the ENG-13560 in this comment as well?

File: src/frontend/org/voltdb/importer/ImporterLifeCycleManager.java [hunk 3]
```diff
@@ -268,12 +268,12 @@ public final void stop()
         }
 
         if (m_executorService != null) {
-            m_executorService.shutdown();
+            //shutdown right away to avoid delay. Importers should pick up where they left.
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/importer/ImporterLifeCycleManager.java]: @zeemanhe Can you add the ENG-13560 in this comment as well?

Comment (zeemanhe) [reply src/frontend/org/voltdb/importer/ImporterLifeCycleManager.java]: sure

**Comment by xinjiacs** [comment]
@zeemanhe I verified that this issue starts from kafka10 upgrade. 
The ImporterLifeCycleManager awaitTermination was there since 2015. It's only the kafka10 client not getting shutting down immediately that blocks the process.

**Pull request merged**